src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -235,27 +241,44 @@ internal void SetExpired(EvictionReason reason)
             DetachTokens();
         }
 
-        private bool CheckForExpiredTime(DateTimeOffset now)
+        private bool CheckForExpiredTime(in DateTimeOffset now)
         {
-            if (_absoluteExpiration.HasValue && _absoluteExpiration.Value <= now)
+            if (!_absoluteExpiration.HasValue && !_absoluteExpiration.HasValue)
```

Comment:
Should one of these be `_slidingExpiration` instead of `_absoluteExpiration`?

Reply:
yes! I have no idea how I've missed that (despite Eric comment)